install/terraform/modules/gke/cluster.tf

```diff
@@ -56,7 +60,7 @@ resource "null_resource" "test-setting-variables" {
 
 resource "google_container_cluster" "primary" {
   name       = local.name
-  location   = local.zone
+  location   = local.zone == "" ? local.location : local.zone
```

Comment:
I would make this prioritize location.

It looks like warnings are not straightforward, so I think skipping that is fine. But at the very least, let's add an error if they're both non-empty and mismatched. I think it can be done like: https://medium.com/codex/terraform-variable-validation-b9b3e7eddd79

Later we can start erroring when `zone` is provided.

Reply:
If we make the location a priority, do we still want location to have a default value? This would also effect the validation done on the variable. 

As for cross variable validation, it's currently not supported but there seems to be hacky ways around it if we really need the check.

Reply:
So zone no longer has a default and will only be used as the location if zone is not empty.